includes/class-freemius.php

```diff
@@ -9659,59 +9697,93 @@ private function get_installs_data_for_api(
 
             $sites = self::get_sites();
 
+            $site_info_by_install_id_map = array();
+            $duplicate_install_id_map    = array();
+
             foreach ( $sites as $site ) {
                 $blog_id = self::get_site_blog_id( $site );
 
                 $install = $this->get_install_by_blog_id( $blog_id );
 
-                if ( is_object( $install ) ) {
-                    if ( $install->user_id != $this->_user->id ) {
-                        // Install belongs to a different owner.
-                        continue;
-                    }
+                if ( ! is_object( $install ) ) {
+                    continue;
+                }
 
-                    if ( ! $this->is_premium() && $install->is_tracking_prohibited() ) {
-                        // Don't send updates regarding opted-out installs.
-                        continue;
-                    }
+                if ( isset( $site_info_by_install_id_map[ $install->id ] ) ) {
+                    /**
+                     * If the install already exists in the map, skip it as it's a clone that has not been resolved automatically. Clones like this one shouldn't be updated and should be resolved manually via the manual clone resolution admin notice.
+                     *
+                     * @author Leo Fajardo (@leorw)
+                     * @since 2.5.0
+                     */
+                    $duplicate_install_id_map[ $install->id ] = true;
+                    continue;
+                }
 
-                    $install_data = $this->get_site_info( $site );
+                if ( $install->user_id != $this->_user->id ) {
+                    // Install belongs to a different owner.
+                    continue;
+                }
 
-                    $uid = $install_data['uid'];
+                if ( ! $this->is_premium() && $install->is_tracking_prohibited() ) {
+                    // Don't send updates regarding opted-out installs.
+                    continue;
+                }
 
-                    unset( $install_data['blog_id'] );
-                    unset( $install_data['uid'] );
+                $site_info_by_install_id_map[ $install->id ] = array(
+                    'site'    => $site,
+                    'install' => $install,
+                );
+            }
 
-                    $install_data['is_disconnected'] = $install->is_disconnected;
-                    $install_data['is_active']       = $this->is_active_for_site( $blog_id );
-                    $install_data['is_uninstalled']  = $install->is_uninstalled;
+            foreach ( $site_info_by_install_id_map as $site_info ) {

```

Comment:
@vovafeldman Its purpose is to prevent executing `$install = $this->get_install_by_blog_id( $blog_id );` more than once in case of duplicates since it can load the options table of more than one blog only to find the same install. The other purpose is to avoid removing entries from the `$installs_data` array if later on, we find a duplicate.